doom-emacs/.config/doom/config.org
#+TITLE: config.org

* Table of contents :TOC:
- [[#about][about]]
  - [[#doom-emacs][doom emacs]]
  - [[#notes][notes]]
  - [[#files][files]]
- [[#common][common]]
  - [[#variables-and-functions-for-org-files][variables and functions for org files]]
  - [[#user-name-and-email][user name and email]]
  - [[#line-number][line number]]
  - [[#hooks][hooks]]
  - [[#tex-engine-using-xetex][TeX engine (using xetex)]]
  - [[#org-what-programs-to-open-file][org what programs to open file]]
  - [[#org-todo-keywords-faces][org todo-keywords, faces]]
  - [[#proxy-settings][proxy settings]]
- [[#ui][UI]]
  - [[#font][font]]
  - [[#theme][theme]]
- [[#packages][packages]]
  - [[#org-capture][org-capture]]
  - [[#org-roam][org-roam]]
  - [[#org-roam-server][org-roam-server]]
  - [[#org-re-reveal][org-re-reveal]]
  - [[#simplenote][simplenote]]
  - [[#deft][deft]]

* about
** doom emacs
[[https://github.com/hlissner/doom-emacs][doom-emacs]] github repo

** notes
- [[file:config.el][config.el]] is generated from [[file:config.org][config.org]] by enabling =literate= flag in [[file:init.el][init.el]]

** files
- [[file:config.org][config.org]] (this file) contains various configurations
- [[file:init.el][init.el]] specifies the modules doom loads
- [[file:packages.el][packages.el]] specifies the additional packages you would install

* common
** variables and functions for org files
#+begin_src emacs-lisp :tangle yes
(setq org-directory "~/Nextcloud/org/")
(defvar +org-life-file (expand-file-name "life.org" org-directory)
  "The org file for life.")
(defvar +org-daily-file (expand-file-name "daily.org" org-directory)
  "The org file for daily.")
(defvar +org-goals-file (expand-file-name "goals.org" org-directory)
  "The org file for goals.")
(defvar +org-journal-file (expand-file-name "journal.org" org-directory)
  "The org file for journal.")
(defvar +org-people-file (expand-file-name "people.org" org-directory)
  "The org file for people.")
(defvar +org-reflect-file (expand-file-name "reflect.org" org-directory)
  "The org file for reflect.")
(defvar +org-knowledge-base-dir (expand-file-name "knowledge-base/" org-directory)
  "The org-roam directory for knowledge.")
#+end_src

#+RESULTS:
: +org-knowledge-base-dir

** user name and email
Some functionality uses this to identify you, e.g. GPG configuration, email
clients, file templates and snippets.

#+begin_src emacs-lisp :tangle yes
(setq user-full-name "Alex Lai"
      user-mail-address "alex@alexlai.xyz")
#+end_src

** line number
This determines the style of line numbers in effect. If set to `nil', line
numbers are disabled. For relative line numbers, set this to `relative'.
#+begin_src emacs-lisp :tangle yes
(setq display-line-numbers-type t)
#+end_src

** hooks
#+begin_src emacs-lisp :tangle yes
(add-hook 'org-mode-hook 'auto-fill-mode)
(add-hook 'after-init-hook #'global-emojify-mode)
#+end_src

** TeX engine (using xetex)
#+begin_src emacs-lisp :tangle yes
(setq-default TeX-engine 'xetex)
(setq org-latex-pdf-process '("xelatex -interaction nonstopmode %f"
                              "xelatex -interaction nonstopmode %f"))
#+end_src


** org what programs to open file
#+begin_src emacs-lisp :tangle yes
;; open apps
(setq org-file-apps
      '((auto-mode . emacs)
        (directory . emacs)
        ("\\.mm\\'" . default)
        ("\\.x?html?\\'" . "firefox %s")
        ("\\.pdf\\'" . default)
        ("\\.png\\'" . "sxiv \"%s\"" )
        ("\\.jpg\\'" . "sxiv \"%s\"" )
        ("\\.svg\\'" . "sxiv \"%s\"" )
        ("\\.gif\\'" . "mpv \"%s\""  )
        ("\\.mkv\\'" . "mpv \"%s\""  )
        ("\\.mp3\\'" . "mpv \"%s\""  )
        ("\\.mp4\\'" . "mpv \"%s\""  )
        ))
#+end_src
** org todo-keywords, faces
#+begin_src emacs-lisp :tangle yes
(after! org
  (setq
   org-todo-keywords
   '((sequence "TODO(t)" "DOING(i!)"  "|" "DONE(d!)" "KILL(k!)" )
     (sequence "MOVED(m@/!)" "CANCELED(c@)" "|"))
   org-todo-keyword-faces
   '(("DOING" . "green")
     ("MOVED" . "brown")
     ("KILL" . "grey")
     ("CANCELED" . (:foreground "blue" :weight bold))
     )))
#+end_src
** proxy settings
#+begin_src emacs-lisp :tangle yes
(setq url-proxy-services
   '(("no_proxy" . "^\\(localhost\\|10.*\\)")
     ("http" . "localhost:7890")
     ("https" . "localhost:7890")))
#+end_src

* UI
** font
Doom exposes five (optional) variables for controlling fonts in Doom. Here are the three important ones:

+ `doom-font'
+ `doom-variable-pitch-font'
+ `doom-big-font' -- used for `doom-big-font-mode'; use this for
  presentations or streaming.

They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
font string. You generally only need these two:
#+begin_src emacs-lisp :tangle yes
(setq doom-font (font-spec :family "DejaVuSansMono" :size 14))
#+end_src

** theme
There are two ways to load a theme. Both assume the theme is installed and available. You can either set `doom-theme' or manually load a theme with the
`load-theme' function. This is the default:
#+begin_src emacs-lisp :tangle yes
(setq doom-theme 'doom-nord)
#+end_src

* packages
- functions
  - `load!' for loading external *.el files relative to this one
  - `use-package' for configuring packages
  - `after!' for running code after a package has loaded
  - `add-load-path!' for adding directories to the `load-path', relative to
    this file. Emacs searches the `load-path' when you load packages with
    `require' or `use-package'.
  - `map!' for binding new keys
   
** org-capture
#+begin_src emacs-lisp :tangle yes
(after! org-capture
  (setq org-capture-templates
        '(
          ("d" "daily" entry
           (file+olp+datetree +org-daily-file)
           "* TODO %?\n%i\n%a" :prepend t)
          ("l" "life" entry
           (file+headline +org-life-file "Unfiled")
           "* %?\nCAPTURED: %U\n%i\n%a" :prepend t)
          ("t" "todo(life)" entry
           (file+headline +org-life-file "Unfiled")
           "* TODO %?\nCAPTURED: %U\n%i\n%a" :prepend t)
          ("j" "journal" entry
           (file+olp+datetree +org-journal-file)
           "* %U %?\n%i\n%a")
          ("r" "reflect" entry
           (file+olp+datetree +org-reflect-file)
           "* %U %?\n%i\n%a")
          )
        )
  )
#+end_src

#+RESULTS:
| d | daily | entry | (file+olp+datetree +org-daily-file) | * TODO %? |

** org-roam
#+begin_src emacs-lisp :tangle yes
(use-package org-roam
      :hook
      (after-init . org-roam-mode)
      :custom
      (org-roam-directory +org-knowledge-base-dir))
#+end_src

** org-roam-server
#+begin_src emacs-lisp :tangle yes
;; (use-package org-roam-server
;;   :ensure t
;;   :config
;;   (setq org-roam-server-host "127.0.0.1"
;;         org-roam-server-port 8080
;;         org-roam-server-export-inline-images t
;;         org-roam-server-authenticate nil
;;         org-roam-server-label-truncate t
;;         org-roam-server-label-truncate-length 60
;;         org-roam-server-label-wrap-length 20))
#+end_src

** org-re-reveal
#+begin_src emacs-lisp :tangle yes
(use-package org-re-reveal
  :ensure t
  :config
  (setq org-re-reveal-root "http://cdn.jsdelivr.net/reveal.js/3.0.0/"))
#+end_src

#+RESULTS:
: http://cdn.jsdelivr.net/reveal.js/3.0.0/

** simplenote
#+begin_src emacs-lisp :tangle yes
(require 'simplenote2)
(load! "simplenote.el")
(simplenote2-setup)
(map! :leader
      (:prefix-map ("S" . "simplenote")
       :desc "browse" "b" 'simplenote2-browse
       :desc "new-from-buffer" "n" 'simplenote2-create-note-from-buffer
       :desc "sync" "s" 'simplenote2-sync-notes
       )
      )
(use-package simplenote2
  :config
  (setq simplenote2-markdown-notes-mode 'markdown-mode
        simplenote2-notes-mode 'markdown-mode
        )
  )
#+end_src

#+RESULTS:
: t

(use-package simplenote2
  :ensure t
  :config
  (load-file "~/.config/doom/simplenote.el")
  (simplenote2-setup)
  )

** deft
#+begin_src emacs-lisp :tangle yes
(after! deft
  (setq deft-directory +org-knowledge-base-dir)
  )
#+end_src

#+RESULTS:


vimb/history
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/settings#theme	DuckDuckGo Settings
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/?q=%21yt&t=h_	
https://duckduckgo.com/?q=%21yt&t=h_	
https://duckduckgo.com/l/?kh=-1&uddg=https%3A%2F%2Fwww.youtube.com	
https://www.youtube.com/	YouTube
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/?q=%21w+dwm&t=h_	
https://duckduckgo.com/l/?kh=-1&uddg=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSpecial%3ASearch%3Fsearch%3Ddwm%26go%3DGo	
https://en.wikipedia.org/wiki/Dwm	dwm - Wikipedia
https://warrior.uwaterloo.ca/	UWaterloo Member Portal
https://warrior.uwaterloo.ca/Program/GetProducts?classification=ed824741-7d4d-4bbd-9591-baa4c6a3cfee	Search Programs - UWaterloo Member Portal
https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
http://www.lib.uwaterloo.ca/find.php?sstring=security+computing	Search results - UWaterloo Library
https://primo.tug-libraries.on.ca/primo_library/libweb/action/dlSearch.do?institution=WATERLOO&vid=WATERLOO&onCampus=false&indx=1&dym=true&highlight=true&displayField=title&bulkSize=10&query=any%2Ccontains%2Csecurity+in+computing&search_scope=books_tab&tab=default_tab	Primo by Ex Libris - security in computing
https://primo.tug-libraries.on.ca/primo_library/libweb/action/search.do?cs=frb&ct=frb&frbg=742645821&fctN=facet_frbrgroupid&fctV=742645821&doc=vtug4926730&lastPag=&lastPagIndx=1&rfnGrp=frbr&frbrSrt=date&frbrRecordsSource=Primo+Local&frbrJtitleDisplay=&frbrIssnDisplay=&frbrEissnDisplay=&frbrSourceidDisplay=vtug&query=any%2Ccontains%2Csecurity%20in%20computing&fn=search&indx=1&search_scope=books_tab&dscnt=0&vl(1UIStartWith0)=contains&scp.scps=scope%3A(%22CENTRAL%22)%2Cscope%3A(GUELPH)%2Cscope%3A(LAURIER)%2Cscope%3A(WATERLOO)%2Cscope%3A(WATERLOOALL)%2Cscope%3A(GUELPHALL)%2Cscope%3A(LAURIERALL)%2Cscope%3A(TUGPROCESSALL)&onCampus=false&vid=WATERLOO&institution=WATERLOO&bulkSize=10&ct=search&highlight=true&tab=default_tab&vl(397444793UI0)=any&dym=true&displayField=title&vl(freeText0)=security%20in%20computing&dstmp=1558465555306&fromDL	Primo by Ex Libris - security in computing
https://primo.tug-libraries.on.ca/primo_library/libweb/action/search.do?cs=frb&ct=frb&frbg=742645821&fctN=facet_frbrgroupid&fctV=742645821&doc=vtug4926730&lastPag=&lastPagIndx=1&rfnGrp=frbr&frbrSrt=date&frbrRecordsSource=Primo+Local&frbrJtitleDisplay=&frbrIssnDisplay=&frbrEissnDisplay=&frbrSourceidDisplay=vtug&query=any%2Ccontains%2Csecurity%20in%20computing&fn=search&indx=1&search_scope=books_tab&dscnt=0&vl(1UIStartWith0)=contains&scp.scps=scope%3A(%22CENTRAL%22)%2Cscope%3A(GUELPH)%2Cscope%3A(LAURIER)%2Cscope%3A(WATERLOO)%2Cscope%3A(WATERLOOALL)%2Cscope%3A(GUELPHALL)%2Cscope%3A(LAURIERALL)%2Cscope%3A(TUGPROCESSALL)&onCampus=false&vid=WATERLOO&institution=WATERLOO&bulkSize=10&ct=search&highlight=true&tab=default_tab&vl(397444793UI0)=any&dym=true&displayField=title&vl(freeText0)=security%20in%20computing&dstmp=1558465555306&fromDL	Primo by Ex Libris - security in computing
https://proxy.lib.uwaterloo.ca/login?url=http://proquestcombo.safaribooksonline.com/book/networking/security/9780134085074	
https://primo.tug-libraries.on.ca/primo_library/libweb/action/search.do?cs=frb&ct=frb&frbg=742645821&fctN=facet_frbrgroupid&fctV=742645821&doc=vtug4926730&lastPag=&lastPagIndx=1&rfnGrp=frbr&frbrSrt=date&frbrRecordsSource=Primo+Local&frbrJtitleDisplay=&frbrIssnDisplay=&frbrEissnDisplay=&frbrSourceidDisplay=vtug&query=any%2Ccontains%2Csecurity%20in%20computing&fn=search&indx=1&search_scope=books_tab&dscnt=0&vl(1UIStartWith0)=contains&scp.scps=scope%3A(%22CENTRAL%22)%2Cscope%3A(GUELPH)%2Cscope%3A(LAURIER)%2Cscope%3A(WATERLOO)%2Cscope%3A(WATERLOOALL)%2Cscope%3A(GUELPHALL)%2Cscope%3A(LAURIERALL)%2Cscope%3A(TUGPROCESSALL)&onCampus=false&vid=WATERLOO&institution=WATERLOO&bulkSize=10&ct=search&highlight=true&tab=default_tab&vl(397444793UI0)=any&dym=true&displayField=title&vl(freeText0)=security%20in%20computing&dstmp=1558465555306&fromDL	Primo by Ex Libris - security in computing

https://duckduckgo.com/	DuckDuckGo — Privacy, simplified.
https://duckduckgo.com/?q=what&t=h_	what at DuckDuckGo
https://duckduckgo.com/?q=feh+set+wallpaper+to+one+screen&t=h_&ia=web	feh set wallpaper to one screen at DuckDuckGo
https://duckduckgo.com/l/?kh=-1&uddg=https%3A%2F%2Fwiki.archlinux.org%2Findex.php%2FFeh	
https://wiki.archlinux.org/index.php/Feh	Feh - ArchWiki
https://duckduckgo.com/?q=feh+set+wallpaper+to+one+screen&t=h_&ia=web	feh set wallpaper to one screen at DuckDuckGo
https://bbs.archlinux.org/viewtopic.php?id=215686	Using display variable to change wallpaper of specific monitor / Newbie Corner / Arch Linux Forums
https://duckduckgo.com/?q=feh+set+wallpaper+to+one+screen&t=h_&ia=web	feh set wallpaper to one screen at DuckDuckGo
https://duckduckgo.com/?q=awesome+set+wallpaper&t=h_&ia=web	awesome set wallpaper at DuckDuckGo
https://duckduckgo.com/?q=awesome+wm++set+wallpaper&t=h_&ia=web	awesome wm set wallpaper at DuckDuckGo
https://duckduckgo.com/l/?kh=-1&uddg=https%3A%2F%2Faskubuntu.com%2Fquestions%2F78629%2Fcan-i-set-different-backgrounds-for-different-workspaces-tags-in-awesome-wm
https://askubuntu.com/questions/78629/can-i-set-different-backgrounds-for-different-workspaces-tags-in-awesome-wm	wallpaper - Can I set different backgrounds for different workspaces (tags) in Awesome WM? - Ask Ubuntu
https://duckduckgo.com/?q=awesome+wm+set+wallpaper+for+different+screens&t=h_&ia=web	awesome wm set wallpaper for different screens at DuckDuckGo
https://duckduckgo.com/l/?kh=-1&uddg=https%3A%2F%2Fbbs.archlinux.org%2Fviewtopic.php%3Fid%3D240639	
https://bbs.archlinux.org/viewtopic.php?id=240639	[SOLVED] Awesome WM - How to set different wallpapers on each screen / Newbie Corner / Arch Linux Forums

doom-emacs/.config/doom/config.el
(setq org-directory "~/Nextcloud/org/")
(defvar +org-life-file (expand-file-name "life.org" org-directory)
  "The org file for life.")
(defvar +org-daily-file (expand-file-name "daily.org" org-directory)
  "The org file for daily.")
(defvar +org-goals-file (expand-file-name "goals.org" org-directory)
  "The org file for goals.")
(defvar +org-journal-file (expand-file-name "journal.org" org-directory)
  "The org file for journal.")
(defvar +org-people-file (expand-file-name "people.org" org-directory)
  "The org file for people.")
(defvar +org-reflect-file (expand-file-name "reflect.org" org-directory)
  "The org file for reflect.")
(defvar +org-knowledge-base-dir (expand-file-name "knowledge-base/" org-directory)
  "The org-roam directory for knowledge.")

(setq user-full-name "Alex Lai"
      user-mail-address "alex@alexlai.xyz")

(setq display-line-numbers-type t)

(add-hook 'org-mode-hook 'auto-fill-mode)
(add-hook 'after-init-hook #'global-emojify-mode)

(setq-default TeX-engine 'xetex)
(setq org-latex-pdf-process '("xelatex -interaction nonstopmode %f"
                              "xelatex -interaction nonstopmode %f"))

;; open apps
(setq org-file-apps
      '((auto-mode . emacs)
        (directory . emacs)
        ("\\.mm\\'" . default)
        ("\\.x?html?\\'" . "firefox %s")
        ("\\.pdf\\'" . default)
        ("\\.png\\'" . "sxiv \"%s\"" )
        ("\\.jpg\\'" . "sxiv \"%s\"" )
        ("\\.svg\\'" . "sxiv \"%s\"" )
        ("\\.gif\\'" . "mpv \"%s\""  )
        ("\\.mkv\\'" . "mpv \"%s\""  )
        ("\\.mp3\\'" . "mpv \"%s\""  )
        ("\\.mp4\\'" . "mpv \"%s\""  )
        ))

(after! org
  (setq
   org-todo-keywords
   '((sequence "TODO(t)" "DOING(i!)"  "|" "DONE(d!)" "KILL(k!)" )
     (sequence "MOVED(m@/!)" "CANCELED(c@)" "|"))
   org-todo-keyword-faces
   '(("DOING" . "green")
     ("MOVED" . "brown")
     ("KILL" . "grey")
     ("CANCELED" . (:foreground "blue" :weight bold))
     )))

(setq url-proxy-services
   '(("no_proxy" . "^\\(localhost\\|10.*\\)")
     ("http" . "localhost:7890")
     ("https" . "localhost:7890")))

(setq doom-font (font-spec :family "DejaVuSansMono" :size 14))

(setq doom-theme 'doom-nord)

(after! org-capture
  (setq org-capture-templates
        '(
          ("d" "daily" entry
           (file+olp+datetree +org-daily-file)
           "* TODO %?\n%i\n%a" :prepend t)
          ("l" "life" entry
           (file+headline +org-life-file "Unfiled")
           "* %?\nCAPTURED: %U\n%i\n%a" :prepend t)
          ("t" "todo(life)" entry
           (file+headline +org-life-file "Unfiled")
           "* TODO %?\nCAPTURED: %U\n%i\n%a" :prepend t)
          ("j" "journal" entry
           (file+olp+datetree +org-journal-file)
           "* %U %?\n%i\n%a")
          ("r" "reflect" entry
           (file+olp+datetree +org-reflect-file)
           "* %U %?\n%i\n%a")
          )
        )
  )

(use-package org-roam
      :hook
      (after-init . org-roam-mode)
      :custom
      (org-roam-directory +org-knowledge-base-dir))

;; (use-package org-roam-server
;;   :ensure t
;;   :config
;;   (setq org-roam-server-host "127.0.0.1"
;;         org-roam-server-port 8080
;;         org-roam-server-export-inline-images t
;;         org-roam-server-authenticate nil
;;         org-roam-server-label-truncate t
;;         org-roam-server-label-truncate-length 60
;;         org-roam-server-label-wrap-length 20))

(use-package org-re-reveal
  :ensure t
  :config
  (setq org-re-reveal-root "http://cdn.jsdelivr.net/reveal.js/3.0.0/"))

(require 'simplenote2)
(load! "simplenote.el")
(simplenote2-setup)
(map! :leader
      (:prefix-map ("S" . "simplenote")
       :desc "browse" "b" 'simplenote2-browse
       :desc "new-from-buffer" "n" 'simplenote2-create-note-from-buffer
       :desc "sync" "s" 'simplenote2-sync-notes
       )
      )
(use-package simplenote2
  :config
  (setq simplenote2-markdown-notes-mode 'markdown-mode
        simplenote2-notes-mode 'markdown-mode
        )
  )

(after! deft
  (setq deft-directory +org-knowledge-base-dir)
  )
